File: Projects/RecruitmentSystem/backend/recruitment-service/src/main/java/com/recruitment/recruitmentservice/entity/Job.java
package com.recruitment.recruitmentservice.entity;

import com.recruitment.recruitmentservice.entity.enums.EmploymentType;
import com.recruitment.recruitmentservice.entity.enums.ExperienceLevel;
import com.recruitment.recruitmentservice.entity.enums.JobStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "jobs")
public class Job extends BaseEntity {

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, unique = true, length = 50)
    private String jobCode;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String requirements;

    @Column(columnDefinition = "TEXT")
    private String responsibilities;

    @Column(precision = 15, scale = 2)
    private BigDecimal salaryMin;

    @Column(precision = 15, scale = 2)
    private BigDecimal salaryMax;

    @Column(length = 10)
    private String currency = "VND";

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EmploymentType employmentType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ExperienceLevel experienceLevel;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private JobStatus status = JobStatus.DRAFT;

    @Column(nullable = false)
    private Integer quantity = 1;

    private LocalDate applicationDeadline;

    private LocalDateTime publishedAt;

    private LocalDateTime expiredAt;

    @Column(nullable = false)
    private Boolean remoteAllowed = false;

    @Column(nullable = false)
    private Boolean active = true;

    @Column(nullable = false)
    private UUID companyId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private JobCategory category;

}

File: Projects/RecruitmentSystem/frontend/web-app/src/types/models/job.ts
export const JOB_STATUSES = ['DRAFT', 'PUBLISHED', 'CLOSED', 'EXPIRED'] as const
export const EMPLOYMENT_TYPES = ['FULL_TIME', 'PART_TIME', 'INTERNSHIP', 'FREELANCE', 'CONTRACT', 'TEMPORARY'] as const
export const EXPERIENCE_LEVELS = ['NO_EXPERIENCE', 'FRESHER', 'JUNIOR', 'MIDDLE', 'SENIOR', 'LEADER', 'MANAGER'] as const

export type JobStatus = (typeof JOB_STATUSES)[number]
export type EmploymentType = (typeof EMPLOYMENT_TYPES)[number]
export type ExperienceLevel = (typeof EXPERIENCE_LEVELS)[number]

export type JobSummary = {
  id: string
  title: string
  jobCode: string
  salaryMin: number | null
  salaryMax: number | null
  currency: string | null
  employmentType: EmploymentType | null
  experienceLevel: ExperienceLevel | null
  status: JobStatus
  remoteAllowed: boolean
  location: string | null
  quantity: number | null
  companyId: string
  categoryId: string | null
  categoryName: string | null
  categorySlug: string | null
  applicationDeadline: string | null
  publishedAt: string | null
  active: boolean
}

export type JobDetail = JobSummary & {
  description: string | null
  requirements: string | null
  responsibilities: string | null
  expiredAt: string | null
  createdAt: string
  updatedAt: string
}

export type JobMutationRequest = {
  title: string
  jobCode: string
  description?: string
  requirements?: string
  responsibilities?: string
  salaryMin?: number
  salaryMax?: number
  currency?: string
  employmentType: EmploymentType
  experienceLevel: ExperienceLevel
  quantity?: number
  applicationDeadline?: string
  remoteAllowed?: boolean
  active?: boolean
  companyId: string
  categoryId: string
}

export type JobCategory = {
  id: string
  name: string
  slug: string
  description: string | null
  icon: string | null
  displayOrder: number
  active: boolean
  parentId: string | null
  parentName: string | null
  createdAt: string
  updatedAt: string
}
